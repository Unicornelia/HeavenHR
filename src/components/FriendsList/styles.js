import { fade, makeStyles } from '@material-ui/core/styles'

export const useStyles = makeStyles(theme => ({
  options: {
    display: "flex",
    justifyContent: "space-between"
  },
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.primary.light, 0.15),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    width: "40%",
    [theme.breakpoints.up("sm")]: {}
  },
  searchIcon: {
    padding: theme.spacing(0, 2),
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  inputRoot: {
    color: "inherit"
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("md")]: {
      width: "20ch"
    }
  }
}));