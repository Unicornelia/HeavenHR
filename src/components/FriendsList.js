import React, { useState } from "react";
import * as PropTypes from "prop-types";

import SearchIcon from "@material-ui/icons/Search";
import InputBase from "@material-ui/core/InputBase";
import { fade, makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import Table from "@material-ui/core/Table";

import FriendsTableBody from "./FriendsTableBody";
import TableHeader from "./TableHeader";

const { shape, arrayOf } = PropTypes;

const useStyles = makeStyles(theme => ({
  search: {
    position: "relative",
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.primary.light, 0.15),
    "&:hover": {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    width: "40%",
    [theme.breakpoints.up("sm")]: {}
  },
  searchIcon: {
    padding: theme.spacing(0, 2),
    height: "100%",
    position: "absolute",
    pointerEvents: "none",
    display: "flex",
    alignItems: "center",
    justifyContent: "center"
  },
  inputRoot: {
    color: "inherit"
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
    transition: theme.transitions.create("width"),
    width: "100%",
    [theme.breakpoints.up("md")]: {
      width: "20ch"
    }
  }
}));

const FriendsList = ({ friends }) => {
  const classes = useStyles();
  const [filterText, setFilterText] = useState("");

  return (
    <>
      <Paper className="m-4 mt-0">
        <div className={classes.search}>
          <div className={classes.searchIcon}>
            <SearchIcon />
          </div>
          <InputBase
            placeholder="Searchâ€¦"
            classes={{
              root: classes.inputRoot,
              input: classes.inputInput
            }}
            inputProps={{ "aria-label": "search" }}
            onChange={e => setFilterText(e.target.value)}
          />
        </div>
        <Table className="p-2">
          <TableHeader
            first={"id"}
            second={"name"}
            third={"gender"}
            fourth={"status"}
            fifth={"delete"}
          />
          <FriendsTableBody friends={friends} filterText={filterText} />
        </Table>
      </Paper>
    </>
  );
};

FriendsList.propTypes = {
  friends: arrayOf(shape({}))
};

FriendsList.defaultProps = {
  friends: []
};

export default FriendsList;
