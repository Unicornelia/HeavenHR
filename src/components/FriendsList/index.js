import React, { useState } from "react";
import * as PropTypes from "prop-types";

import SearchIcon from "@material-ui/icons/Search";
import {
  InputBase,
  Paper,
  Table,
  Select,
  MenuItem,
  TableHead,
  TableRow,
  TableCell
} from "@material-ui/core";

import { useStyles } from "./styles";
import FriendsTableBody from "../FriendsTableBody";
import TableHeader from "../TableHeader";
import Pagination from '../Pagination'

const { shape, arrayOf } = PropTypes;

const FriendsList = ({ friends }) => {
  const classes = useStyles();
  const [filterText, setFilterText] = useState("");
  const [gender, setGender] = useState("all");
  const [status, setStatus] = useState("all");
  const [idOrder, setIdOrder] = useState("all");
  const [nameOrder, setNameOrder] = useState("all");
  const [currentPage, setCurrentPage] = useState(1);
  const [itemsPerPage] = useState(5);
  const indexOfLastFriend = currentPage * itemsPerPage;
  const indexOfFirstFriend = indexOfLastFriend - itemsPerPage;

  const paginate = pageNumber => {
    setCurrentPage(pageNumber);
  };

  const handleGenderChange = e => {
    setGender(e.target.value);
  };

  const handleStatusChange = e => {
    setStatus(e.target.value);
  };

  const handleIdOrderChange = e => {
    setIdOrder(e.target.value)
  }

  const handleNameOrderChange = e => {
    setNameOrder(e.target.value)
  }

  return (
    <>
      <Paper className="m-4 mt-0">
        <div className={classes.options}>
          <div className={classes.search}>
            <div className={classes.searchIcon}>
              <SearchIcon />
            </div>
            <InputBase
              placeholder="Searchâ€¦"
              classes={{
                root: classes.inputRoot,
                input: classes.inputInput
              }}
              inputProps={{ "aria-label": "search" }}
              onChange={e => setFilterText(e.target.value)}
            />
          </div>
        </div>
        <Table className="p-2">
          <TableHead>
            <TableRow>
              <TableCell>
                <Select value={idOrder} onChange={handleIdOrderChange}>
                  <MenuItem value={"ascending"}>Ascending</MenuItem>
                  <MenuItem value={"descending"}>Descending</MenuItem>
                  <MenuItem value={"all"}>Sort by ID</MenuItem>
                </Select>
              </TableCell>
              <TableCell>
                <Select value={nameOrder} onChange={handleNameOrderChange}>
                  <MenuItem value={"ascending"}>Ascending</MenuItem>
                  <MenuItem value={"descending"}>Descending</MenuItem>
                  <MenuItem value={"all"}>Sort by name</MenuItem>
                </Select>
              </TableCell>
              <TableCell>
                <Select value={gender} onChange={handleGenderChange}>
                  <MenuItem value={"male"}>Male</MenuItem>
                  <MenuItem value={"female"}>Female</MenuItem>
                  <MenuItem value={"all"}>All</MenuItem>
                </Select>
              </TableCell>
              <TableCell>
                <Select value={status} onChange={handleStatusChange}>
                  <MenuItem value={true}>True</MenuItem>
                  <MenuItem value={false}>False</MenuItem>
                  <MenuItem value={"all"}>All</MenuItem>
                </Select>
              </TableCell>
              <TableCell className="text-uppercase" />
            </TableRow>
          </TableHead>
          <TableHeader />
          <FriendsTableBody
            friends={friends}
            filterText={filterText}
            selectedGender={gender}
            selectedStatus={status}
            selectedIdOrder={idOrder}
            selectedNameOrder={nameOrder}
            indexOfFirstFriend={indexOfFirstFriend}
            indexOfLastFriend={indexOfLastFriend}
          />
        </Table>
        <Pagination
          itemsPerPage={itemsPerPage}
          total={friends.length}
          paginate={paginate}
        />
      </Paper>
    </>
  );
};

FriendsList.propTypes = {
  friends: arrayOf(
    shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      gender: PropTypes.string.isRequired,
      isStarred: PropTypes.bool.isRequired
    })
  )
};

FriendsList.defaultProps = {
  friends: []
};

FriendsList.displayName = "FriendsList";

export default FriendsList;
