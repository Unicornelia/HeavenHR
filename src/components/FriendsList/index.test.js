import React from "react";
import configureStore from "redux-mock-store";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure, mount } from "enzyme";
import { Provider } from "react-redux";
import faker from "faker";

import FriendsList from "./index";
configure({ adapter: new Adapter() });

describe("<FriendsList/>", () => {
  const mockStore = configureStore();
  const initialState = [
    {
      id: faker.random.uuid(),
      name: faker.name.findName(),
      gender: "male",
      isStarred: false
    },
    {
      id: faker.random.uuid(),
      name: faker.name.findName(),
      gender: "female",
      isStarred: false
    }
  ];
  const store = mockStore(initialState);

  it("should match snapshot", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <FriendsList friends={initialState}/>
      </Provider>
    );

    const component = wrapper.dive();

    expect(component).toMatchSnapshot();
    expect(store.getActions()).toMatchSnapshot();
  });

  it("should find friends table body", () => {
    const wrapper = mount(
      <Provider store={store}>
        <FriendsList friends={initialState}/>
      </Provider>
    );

    expect(wrapper.find('FriendsTableBody')).toHaveLength(1)
  });
});
