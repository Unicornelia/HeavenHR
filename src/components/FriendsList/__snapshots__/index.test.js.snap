// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FriendsList/> should match snapshot 1`] = `ShallowWrapper {}`;

exports[`<FriendsList/> should match snapshot 2`] = `Array []`;
